File: CMPA/OptiLightData.cpp
// 
// 
// 

#include "OptiLightData.h"

OptiLightData::OptiLightData(double vLow , double vHigh)
{
	VLow = vLow;
	VHigh = vHigh;
	EstimatedFETopenTime = 1;
	EstimatedFETcloseTime = 2;
	Tries = 0;
	HighestAcceleration = 0;
}

OptiLightData::OptiLightData(double vLow, int vHigh, double estimatedFETopenTime, double estimatedFETcloseTime, double highestAcceleration, int tries)
{
	VLow = vLow;
	VHigh = vHigh;
	EstimatedFETopenTime = estimatedFETopenTime;
	EstimatedFETcloseTime = estimatedFETcloseTime;
	HighestAcceleration = highestAcceleration;
	Tries = tries;
}

String OptiLightData::ToString()
{
	return VLow + String(";") + VHigh + ";" + EstimatedFETopenTime + ";" + EstimatedFETcloseTime + ";" + HighestAcceleration + ";" + Tries;
}

File: OptiCom/Modelling/OptiLightData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace OptiCom.Modelling
{
    public class OptiLightData
    {
        public readonly double VLow;
        public readonly double VHigh;
        public readonly double EstimatedFETOffTime;

        public OptiLightData(double vLow, double vHigh, double estimatedFETOffTime)
        {
            VLow = vLow;
            VHigh = vHigh;
            EstimatedFETOffTime = estimatedFETOffTime;
        }

        public override string ToString()
        {
            //return OptiLightGenerator.MAGIC_NUMBER + ";" + VLow.ToString("0.0") + ";" + VHigh.ToString("0.0") + ";" + EstimatedFETOffTime.ToString("0.00000") + ";0;0";
            return VLow.ToString("0.0", System.Globalization.CultureInfo.InvariantCulture) + "," + VHigh.ToString("0.0", System.Globalization.CultureInfo.InvariantCulture) + "," + EstimatedFETOffTime.ToString("0.00000", System.Globalization.CultureInfo.InvariantCulture);
        }
    }
}
